src/types/select.js

```diff
@@ -797,17 +797,17 @@ export default {
 	},
 
 	deMap: function (obj) {
-		if(!obj){
+		if (!obj) {
 			return null;
 		}
-		return obj.value || null;
+		return obj.value;
 	},
 
 	isEmpty: function () {
 		const self = this,
 			data = self.deMap(self.item);
 
-		return data !== 0 && !data;
+		return data === 0 || data === undefined || data === null || data === '';
```

Comment:
this should be changed to: 
`return data !== 0 && data !== false && !data;`
So that the only change is treating the falsey value of `false` as a set value along with `0`.